track_info.c

```diff
@@ -183,6 +186,55 @@ static int doublecmp0(double a, double b)
 	return (x > 0) - (x < 0);
 }
 
+int strnatcmp(const char * str1, const char * str2)
+{
+	int result;
+	char *o_ptr1, *o_ptr2, *strcpy1, *strcpy2;
+
+	o_ptr1 = xstrndup(str1, strlen(str1) + 1);
+	o_ptr2 = xstrndup(str2, strlen(str2) + 1);
```

Comment:
I don't see any real need to have `xstrndup`/`free` calls in `strnatcmp`. This is a comparator, and it must be fast (including not allocating memory).